### PR TITLE
[Flask] Uninstall snap

### PR DESCRIPTION
**Pull request by owencraston**
**Development & PR Process**
1. Follow MetaMask [Mobile Coding Standards](https://docs.google.com/document/d/1VJLwTRsUw_5EDq_o8d6sSbXUAYENLurkRitYO45eY5o/edit?usp=sharing)
2. Add `release-xx` label to identify the PR slated for a upcoming release (will be used in release discussion)
3. Add `needs-dev-review` label when work is completed
4. Add `needs-qa` label when dev review is completed
5. Add `QA Passed` label when QA has signed off

**Description**

_Write a short description of the changes included in this pull request, also include relevant motivation and context. Have in mind the following questions,_
_1. What is the reason for the change?_
- we want to show all the installed snaps, even the ones that were installed in a previous session
- we want the ability to uninstall a snap
_2. What is the improvement/solution?_
- pass in the correct initial state (the one that is persisted in the database) to the snaps controller so that we can see the history of installed snaps
- then we call SnapsController.removeSnap to uninstall a specified snap

**Screenshots/Recordings**

https://github.com/MetaMask/metamask-mobile/assets/22918444/7f19ee5c-870b-4ca3-8e8b-b7441d7ce67f

**Issue**

Progresses https://github.com/MetaMask/mobile-planning/issues/745
Progresses https://github.com/MetaMask/mobile-planning/issues/803

**Checklist**

* [ ] There is a related GitHub issue
* [ ] Tests are included if applicable
* [ ] Any added code is fully documented

### CODE REVIEW

File: app/components/Views/Snaps/SnapElement/SnapElement.tsx [hunk 1]
```diff
@@ -22,9 +22,9 @@ const SnapElement = (snap: any) => {
     console.log('ping');
   };
 
-  const terminate = async () => {
+  const removeSnap = async () => {
     const { SnapController } = Engine.context as any;
-    await SnapController.terminateSnap(snap.snap.id);
+    await SnapController.removeSnap(snap.snap.id);
   };
```
Comment (gantunesr) [inline app/components/Views/Snaps/SnapElement/SnapElement.tsx:28]: I know this is just an UI/UX used for development, but we need to make a distinction between "Terminate" and "Remove". Flask currently allows the user to Uninstall a snap (Remove) and stop it (Terminate)

**Review by gantunesr — APPROVED** [review]

File: app/components/Views/Snaps/SnapElement/SnapElement.tsx (hunk 1 above)
Comment (owencraston) [reply app/components/Views/Snaps/SnapElement/SnapElement.tsx:28]: I will add both for now.

**Comment by owencraston** [comment]
I was able to stop the snap (stop but not remove) by simply swapping the `terminateSnap` method with the `stopSnap` method

https://github.com/MetaMask/metamask-mobile/assets/22918444/846db7ac-17d1-4bfc-8c92-b0c841b63e54

**Comment by owencraston** [comment]
Logs after stopping a snap

```
 LOG  [SNAP DUPLEX LOG] SnapDuplex+_write: Job 6sZd62_K3uBi9F6ylNwEv
 LOG  [APP LOG] setWebviewPostMessage: Message from Webview [object Object]
 LOG  [SNAP DUPLEX LOG] SnapDuplex+_onData: Job 6sZd62_K3uBi9F6ylNwEv read data {"data": {"data": {"id": "x3A4rbfKwEoxORjjEZ-oK", "jsonrpc": "2.0", "result": "OK"}, "name": "command"}, "jobId": "6sZd62_K3uBi9F6ylNwEv"}
 LOG  [SNAP DUPLEX LOG] SnapDuplex+destroy: Destroy stream from SnapDuplex
 LOG  [WebviewPostMessageStream LOG]: Destroy stream
 LOG  [SNAP DUPLEX LOG] SnapDuplex+destroy: Destroy stream from SnapDuplex
 LOG  [WebviewPostMessageStream LOG]: Destroy stream
 LOG  [EXEC SERVICE LOG] WebviewExecutionService+_terminate: Job 6sZd62_K3uBi9F6ylNwEv SnapDuplex destroyed
 LOG  Job "6sZd62_K3uBi9F6ylNwEv" terminated.
 ```

Logs after removing a snap

```
 LOG  Parent: Sending Command {"id": "MTRA2pJOZbM2XJVBFQhdO", "jsonrpc": "2.0", "method": "terminate", "params": []}
 LOG  [SNAP DUPLEX LOG] SnapDuplex+_write: Job xb61hy6g82LLaM-ymht1w
 LOG  [APP LOG] setWebviewPostMessage: Message from Webview [object Object]
 LOG  [SNAP DUPLEX LOG] SnapDuplex+_onData: Job xb61hy6g82LLaM-ymht1w read data {"data": {"data": {"id": "MTRA2pJOZbM2XJVBFQhdO", "jsonrpc": "2.0", "result": "OK"}, "name": "command"}, "jobId": "xb61hy6g82LLaM-ymht1w"}
 LOG  [SNAP DUPLEX LOG] SnapDuplex+destroy: Destroy stream from SnapDuplex
 LOG  [WebviewPostMessageStream LOG]: Destroy stream
 LOG  [SNAP DUPLEX LOG] SnapDuplex+destroy: Destroy stream from SnapDuplex
 LOG  [WebviewPostMessageStream LOG]: Destroy stream
 LOG  [EXEC SERVICE LOG] WebviewExecutionService+_terminate: Job xb61hy6g82LLaM-ymht1w SnapDuplex destroyed
 LOG  Job "xb61hy6g82LLaM-ymht1w" terminated.
 ```

**Pull request merged**